### PR TITLE
[camera_web] Release the camera stream of each available video input device

### PR DESCRIPTION
**Pull request by bselwe**
Accessing available cameras fetches the video stream of each video input device from [MediaDevices.enumerateDevices](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices). Those video streams are now released when the camera metadata is created. This should solve a `cameraNotReadable` error thrown when initializing a camera that is already in use.

The solution was proposed by @leandroveronezi in https://github.com/flutter/flutter/issues/90881#issuecomment-988072447, thanks!

Closes https://github.com/flutter/flutter/issues/90881.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by bselwe** [comment]
@ditman

**Review by ditman — APPROVED** [review]
LGTM, let's give this a shot!

**Comment by ditman** [comment]
_(PS: the `availableCameras` method is starting to get a little bit unwieldy (100+ LoC), I wonder if there's any good logical split for all the things that happen inside of it!)_

File: packages/camera/camera_web/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.2.1+2
+
+* Fix cameraNotReadable error that prevented access to the camera on some Android devices when initializing a camera.
```
Comment (stuartmorgan) [inline packages/camera/camera_web/CHANGELOG.md:3]: Nit: Fixes

**Review by stuartmorgan — APPROVED** [review]
Looks like this fell through the cracks on getting a second reviewer, sorry about that. LGTM with one nit.

File: packages/camera/camera_web/CHANGELOG.md (hunk 1 above)
Comment (ditman) [reply packages/camera/camera_web/CHANGELOG.md]: Done!

**Comment by bselwe** [comment]
Thanks @ditman for taking care of this one!

**Comment by ditman** [comment]
No problem @bselwe! Thanks for the OG fix!

**Pull request merged**